src/explore-education-statistics-frontend/src/prototypes/components/PrototypeSearchForm.tsx

```diff
@@ -138,6 +134,80 @@ class PrototypeSearchForm extends Component<{}, State> {
     this.setState({ searchResults });
   }
 
+  private static openTab(target: Element, container: Element) {
+    // find the panel in which the element lives
+    const panel = PrototypeSearchForm.parentUntilClassname(
+      target,
+      'govuk-tabs__panel',
+    );
+
+    // find all the sections within the container
+    const allSections = container.querySelectorAll('.govuk-tabs__panel');
+
+    // get the index of the panel within the container
+    const selectedPanel = Array.from(allSections).indexOf(panel);
+
+    // only proceed if all this is valid
+    if (selectedPanel >= 0) {
+      // find the selected anchor
+      const anchor = container
+        .querySelectorAll('ul.govuk-tabs__list > li.govuk-tabs__list-item')
+        .item(selectedPanel)
+        .querySelector('a');
+
+      if (anchor) {
+        // Deselect all the selected tabs, make sure we found the anchor first
+        container
+          .querySelectorAll(
+            'ul.govuk-tabs__list > li.govuk-tabs__list-item a.govuk-tabs__tab--selected',
+          )
+          .forEach(_ => {
+            _.classList.remove('govuk-tabs__tab--selected');
+            _.setAttribute('aria-selected', 'false');
+            _.setAttribute('tabIndex', '-1');
+          });
+
+        // make the selected tab selected
+        anchor.classList.add('govuk-tabs__tab--selected');
+        anchor.setAttribute('aria-selected', 'true');
+        anchor.removeAttribute('tabIndex');
+
+        // hide all the panels
+        Array.from(
+          container.querySelectorAll(
+            '.govuk-tabs__panel:not(.govuk-tabs__panel--hidden)',
+          ),
+        ).forEach(_ => _.classList.add('govuk-tabs__panel--hidden'));
+
+        // show the selected panel
+        panel.classList.remove('govuk-tabs__panel--hidden');
+
+        // click the anchor to get its component to update any internal state
+        anchor.dispatchEvent(
+          new MouseEvent('click', {
+            view: window,
+            bubbles: true,
+            cancelable: true,
+            buttons: 1,
+          }),
+        );
+      }
+    }
+  }
+
+  private static openDetails(collapsedContainer: Element) {
+    collapsedContainer.setAttribute('open', '');
+    const textEl = collapsedContainer.querySelector('.govuk-details__text');
+    if (textEl) {
+      textEl.setAttribute('aria-hidden', 'false');
```

Comment:
Whilst the implementation here might be okay for now, I'm a little worried about these various non-React driven DOM manipulations causing React and the DOM to get out of sync.

I think we should consider using a more React friendly way of doing this such as either MobX or Redux before this component graduates into a finalised state.

Reply:
This is one of the GovUK components.